### PR TITLE
[WIP] Try to make iOS CI happy

### PR DESCRIPTION
**Pull request by siddarthkay**
status : wip

### CODE REVIEW

**Comment by siddarthkay** [comment]
IOS seems to be happy now, lets add more changes

File: ios/StatusIm.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1112,7 +1112,7 @@
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				ALWAYS_SEARCH_USER_PATHS = NO;
 				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
-				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
+				CLANG_CXX_LANGUAGE_STANDARD = "c++17";
```
Comment (jakubgs) [inline ios/StatusIm.xcodeproj/project.pbxproj:1115]: So we're changing the C++ standard used by compiler, and this is quite a big leap.

According to this:
```
standard          with gnu extensions

c++98             gnu++98
c++03             gnu++03
c++11 (c++0x)     gnu++11 (gnu++0x)
c++14 (c++1y)     gnu++14 (gnu++1y)
```

https://stackoverflow.com/questions/26822712/whats-the-point-of-including-std-c0x-in-a-g-compile-command

Which means we are jumping from essentially `c++11` to `c++17`. Which means we are essentially leaping 6 years forward in the standards timeline:
https://www.geeksforgeeks.org/c-11-vs-c-14-vs-c-17/

I wonder if `c++14` could do the same thing for us? Also, what error exactly are we trying to fix?

Comment (jakubgs) [reply ios/StatusIm.xcodeproj/project.pbxproj:1115]: I'm assuming this?
```
node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedTurboModule.mm:18:44: error: cannot find protocol declaration for 'NativeAnimatedModuleSpec'
@interface RCTNativeAnimatedTurboModule() <NativeAnimatedModuleSpec, RCTInitializing>
                                           ^
```

Comment (siddarthkay) [reply ios/StatusIm.xcodeproj/project.pbxproj:1115]: Good question @jakubgs 
I am using this PR to just redo the things i did here : https://github.com/status-im/status-mobile/pull/15645 
but in a different order..
I'm trying to add small changes to the CI to see if breaks CI or not.

Comment (siddarthkay) [reply ios/StatusIm.xcodeproj/project.pbxproj:1115]: I also saw this change in react-native upgrade helper here : https://react-native-community.github.io/upgrade-helper/?from=0.67.5&to=0.69.6

**Comment by siddarthkay** [comment]
ah welcome old friend 

```log
[2023-04-14T09:48:21.135Z] [12:48:20]: ▸ ❌  /Users/jenkins/workspace/status-mobile_prs_ios_PR-15649/node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedModule.mm:224:33: use of undeclared identifier 'JS'

[2023-04-14T09:48:21.135Z] [12:48:20]: ▸                   eventMapping:(JS::NativeAnimatedModule::EventMapping &)eventMapping)

[2023-04-14T09:48:21.135Z] [12:48:20]: ▸        ^

[2023-04-14T09:48:21.135Z] [12:48:20]: ▸ ❌  /Users/jenkins/workspace/status-mobile_prs_ios_PR-15649/node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedModule.mm:227:76: property 'nativeEventPath' not found on object of type '__strong id'

[2023-04-14T09:48:21.135Z] [12:48:20]: ▸   eventMappingDict[@"nativeEventPath"] = RCTConvertVecToArray(eventMapping.nativeEventPath());

[2023-04-14T09:48:21.135Z] [12:48:20]: ▸                                 ^

[2023-04-14T09:48:21.135Z] [12:48:20]: ▸ ❌  /Users/jenkins/workspace/status-mobile_prs_ios_PR-15649/node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedModule.mm:229:20: property 'animatedValueTag' not found on object of type '__strong id'

[2023-04-14T09:48:21.136Z] [12:48:20]: ▸   if (eventMapping.animatedValueTag()) {

[2023-04-14T09:48:21.136Z] [12:48:20]: ▸                                                                            ^

[2023-04-14T09:48:21.136Z] [12:48:20]: ▸ ❌  /Users/jenkins/workspace/status-mobile_prs_ios_PR-15649/node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedModule.mm:230:61: property 'animatedValueTag' not found on object of type '__strong id'

[2023-04-14T09:48:21.136Z] [12:48:20]: ▸     eventMappingDict[@"animatedValueTag"] = @(*eventMapping.animatedValueTag());

[2023-04-14T09:48:21.136Z] [12:48:20]: ▸     
```

**Comment by flexsurfer** [comment]
feel free to reopen if still relevant and you're currently working on it

**Pull request closed without merging**